extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcClientProcessor.java

```diff
@@ -345,16 +346,21 @@ public void transform(TransformationContext ctx) {
                 AnnotationInstance clientAnnotation = Annotations.find(ctx.getQualifiers(), GrpcDotNames.GRPC_CLIENT);
                 if (clientAnnotation != null && clientAnnotation.value() == null) {
                     String clientName = null;
+                    AnnotationTarget annotationTarget = ctx.getTarget();
                     if (ctx.getTarget().kind() == Kind.FIELD) {
                         clientName = clientAnnotation.target().asField().name();
-                    } else if (ctx.getTarget().kind() == Kind.METHOD_PARAMETER) {
+                    } else if (ctx.getTarget().kind() == Kind.METHOD
+                            && clientAnnotation.target().kind().equals(Kind.METHOD_PARAMETER)) {
                         MethodParameterInfo param = clientAnnotation.target().asMethodParameter();
+                        annotationTarget = param;
                         // We don't need to check if parameter names are recorded - that's validated elsewhere
                         clientName = param.method().parameterName(param.position());
                     }
                     if (clientName != null) {
                         ctx.transform().remove(GrpcDotNames::isGrpcClient)
-                                .add(GrpcDotNames.GRPC_CLIENT, AnnotationValue.createStringValue("value", clientName)).done();
+                                .add(AnnotationInstance.create(GrpcDotNames.GRPC_CLIENT, annotationTarget,
+                                        new AnnotationValue[] { AnnotationValue.createStringValue("value", clientName) }))
```

Comment:
For your consideration, the `AnnotationInstanceBuilder` is a much nicer API.

```java
AnnotationInstance.builder(GrpcDotNames.GRPC_CLIENT)
    .value(clientName)
    .buildWithTarget(annotationTarget)
```

Reply:
Ah, I didn't realize there's a builder. Thanks!